subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/locking/DependencyLockingPluginIntegrationTest.groovy

```diff
@@ -0,0 +1,329 @@
+}
+
+repositories {
+    maven {
+        name 'repo'
+        url '${mavenRepo.uri}'
+    }
+}
+configurations {
+    lockedConf
+}
+
+dependencies {
+    constraints {
+        lockedConf('org:foo') {
+            version {
+                prefer '1.1'
+            }
+        }
+    }
+    lockedConf 'org:foo:1.+'
+}
+"""
+
+        lockfileFixture.createLockfile('lockedConf',['org:foo:1.0'])
+
+        when:
+        fails 'dependencies'
+
+        then:
+        failure.assertHasCause("Cannot find a version of 'org:foo' that satisfies the version constraints: \n" +
+            "   Dependency path ':depLock:unspecified' --> 'org:foo' prefers '1.+'\n" +
+            "   Constraint path ':depLock:unspecified' --> 'org:foo' prefers '1.1'\n" +
+            "   Constraint path ':depLock:unspecified' --> 'org:foo' prefers '1.0', rejects ']1.0,)' because of the following reason: dependency-locking in place")
+    }
+
+    @Ignore("Post resolve validation missing")
+    def 'fails when lock file entry not resolved'() {
+        mavenRepo.module('org', 'foo', '1.0').publish()
+        mavenRepo.module('org', 'bar', '1.0').publish()
+
+        buildFile << """
+dependencyLocking {
+    lockAllConfigurations()
+}
+
+repositories {
+    maven {
+        name 'repo'
+        url '${mavenRepo.uri}'
+    }
+}
+configurations {
+    lockedConf
+}
+
+dependencies {
+    lockedConf 'org:foo:1.+'
+}
+"""
+
+        lockfileFixture.createLockfile('lockedConf',['org:bar:1.0', 'org:foo:1.0'])
+
+        when:
+        fails 'dependencies'
+
+        then:
+        failure.assertThatCause(containsText("Lock file contained 'org:bar:1.0' but it is not part of the resolved modules"))
+    }
+
+    def 'writes dependency lock file when requested'() {
+        mavenRepo.module('org', 'foo', '1.0').publish()
+        mavenRepo.module('org', 'bar', '1.0').publish()
+
+        buildFile << """
+dependencyLocking {
+    lockAllConfigurations()
+}
+
+repositories {
+    maven {
+        name 'repo'
+        url '${mavenRepo.uri}'
+    }
+}
+configurations {
+    lockedConf
+}
+
+dependencies {
+    lockedConf 'org:foo:1.+'
+    lockedConf 'org:bar:1.+'
+}
+"""
+
+        when:
+        succeeds'dependencies', '--write-locks'
+
+        then:
+        lockfileFixture.verifyLockfile('lockedConf', ['org:bar:1.0', 'org:foo:1.0'])
+
+    }
+
+    def 'upgrades lock file'() {
+        mavenRepo.module('org', 'foo', '1.0').publish()
+        mavenRepo.module('org', 'foo', '1.1').publish()
+
+        buildFile << """
+dependencyLocking {
+    lockAllConfigurations()
+}
+
+repositories {
+    maven {
+        name 'repo'
+        url '${mavenRepo.uri}'
+    }
+}
+configurations {
+    lockedConf
+}
+
+dependencies {
+    constraints {
+        lockedConf('org:foo') {
+            version {
+                prefer '1.1'
+            }
+        }
+    }
+    lockedConf 'org:foo:1.+'
+}
+"""
+
+        file('gradle', 'dependency-locks', 'lockedConf.lockfile') << """
```

Comment:
Should probably use the test fixture here.

Reply:
Indeed, updated